monai/networks/layers/simplelayers.py

```diff
@@ -35,44 +36,44 @@ def forward(self, x):
         return x.view(x.size(0), -1)
 
 
-class GaussianFilter:
-    def __init__(self, spatial_dims, sigma, truncated=4.0, device=None):
+class GaussianFilter(nn.Module):
+    def __init__(self, spatial_dims, sigma, truncated=4.0):
         """
         Args:
             spatial_dims (int): number of spatial dimensions of the input image.
                 must have shape (Batch, channels, H[, W, ...]).
-            sigma (float): std.
+            sigma (float or sequence of floats): std.
             truncated (float): spreads how many stds.
-            device (torch.device): device on which the tensor will be allocated.
         """
-        self.kernel = torch.nn.Parameter(torch.tensor(gaussian_1d(sigma, truncated)), False)
-        self.spatial_dims = spatial_dims
+        super().__init__()
+        self.spatial_dims = int(spatial_dims)
+        _sigma = ensure_tuple_rep(sigma, self.spatial_dims)
+        self.kernel = [
+            torch.nn.Parameter(torch.as_tensor(gaussian_1d(s, truncated), dtype=torch.float32), False) for s in _sigma
+        ]
+        self.padding = [same_padding(k.size()[0]) for k in self.kernel]
         self.conv_n = [F.conv1d, F.conv2d, F.conv3d][spatial_dims - 1]
-        self.padding = same_padding(self.kernel.size()[0])
-        self.device = device
-
-        self.kernel = self.kernel.to(self.device)
+        for idx, param in enumerate(self.kernel):
+            self.register_parameter(f"kernel_{idx}", param)
 
     def __call__(self, x):
```

Comment:
As you changed to inherit from `nn.Module`, I think you should overwrite `forward()` instead of `__call__()`, as `__call__()` will call `forward()`: https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L564
Thanks.